### PR TITLE
[BALDE-15] Add Vector(x,y)

### PR DESCRIPTION
**Pull request by 4jeR**
Added basic Vec2 class

### CODE REVIEW

File: src/math/vec2.cpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
+//
+// Created by ajer on 16.12.18.
+//
+
+#include "vec2.h"
+
+namespace Bald::Math {
+
+
+    [[nodiscard]] constexpr Vec2 &Vec2::operator+(const Vec2 &other) noexcept {
```
Comment (BIGbadEL) [inline src/math/vec2.cpp]: constexpr function have to be defined in header file

File: src/math/vec2.cpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+        this->m_x += other.m_x;
```
Comment (BIGbadEL) [inline src/math/vec2.cpp]: operator + can't modify object

File: src/math/vec2.cpp [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        this->m_y += other.m_y;
```
Comment (BIGbadEL) [inline src/math/vec2.cpp]: operator + can't modify object

File: src/math/vec2.cpp [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+        return *this;
```
Comment (BIGbadEL) [inline src/math/vec2.cpp:28]: should return new object

File: src/math/vec2.cpp [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    [[nodiscard]] constexpr Vec2 &Vec2::operator*(double multiplier) noexcept {
+
+        if (multiplier != 0.0) {
```
Comment (BIGbadEL) [inline src/math/vec2.cpp]: why cant't we multiply by 0? and we should use 0.0f notation

File: src/math/vec2.cpp [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+            this->m_y *= multiplier;
```
Comment (BIGbadEL) [inline src/math/vec2.cpp]: operator * shouldn't modify object

File: src/math/vec2.cpp [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+            this->m_x *= multiplier;
+        }
+
+        return *this;
```
Comment (BIGbadEL) [inline src/math/vec2.cpp:68]: should return a new object

File: src/math/vec2.cpp [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+    [[nodiscard]] constexpr Vec2 &Vec2::operator+(const Vec2 &other) noexcept {
+
+        this->m_x += other.m_x;
+        this->m_y += other.m_y;
+
+        return *this;
+    }
+
+    [[nodiscard]] constexpr Vec2 &Vec2::operator*(double multiplier) noexcept {
```
Comment (BIGbadEL) [inline src/math/vec2.cpp]: constexpr function have to be defined in header file

File: src/math/vec2.cpp [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+        if (multiplier != 0.0) {
+            this->m_y *= multiplier;
+            this->m_x *= multiplier;
+        }
+
+        return *this;
+    }
+
+    double Vec2::Len() const {
+        double length = sqrt(pow(this->m_x, 2.0) + pow(this->m_y, 2.0));
```
Comment (BIGbadEL) [inline src/math/vec2.cpp]: we don't need a variable if we just want to return this, it should be just return sqrt...., also consider making it constexpr

File: src/math/vec2.cpp [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+
+        return length;
+    }
+
+
+    double Vec2::ScalarProduct(const Vec2 &other) const {
+        double product = this->m_x * other.m_x + this->m_y * other.m_y;
```
Comment (BIGbadEL) [inline src/math/vec2.cpp]: we don't need that variable it should just return, consider constexpr

File: src/math/vec2.h [hunk 11]
```diff
@@ -0,0 +1,95 @@
+//
+// Created by ajer on 16.12.18.
+//
+
+#pragma once
+#include "pch.h"
```
Comment (BIGbadEL) [inline src/math/vec2.h]: precompiled header should be included in source file, and only if you use libraries included there

File: src/math/vec2.h [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+
+    class Vec2{
+    public:
+        /*
+         * @fn Vec2
+         * @brief constructor
+         * @param [int] x
+         * @param [int] y
+        */
+        Vec2(double x = 0.0, double y = 0.0)
```
Comment (BIGbadEL) [inline src/math/vec2.h]: could be constexpr

File: src/math/vec2.h [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+                :m_x(x), m_y(y){}
+        /*
+         * @fn Vec2
+         * @brief copy constructor
+         * @param [const Vec2&] other vector
+        */
+        Vec2(const Vec2& other)
```
Comment (BIGbadEL) [inline src/math/vec2.h]: if we dont't store pointer and dynamic memory it is unnesessery

File: src/math/vec2.h [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+                :m_x(other.m_x), m_y(other.m_y){}
+
+
+        /*
+         * @fn GetX
+         * @brief returns member X variable
+        */
+        [[nodiscard]] constexpr inline double GetX()const noexcept{return m_x;}
```
Comment (BIGbadEL) [inline src/math/vec2.h]: i would add a faw spaces here

File: src/math/vec2.h [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+         * @fn GetY
+         * @brief returns member Y variable
+        */
+        [[nodiscard]] constexpr inline double GetY()const noexcept{return m_y;}
+
+        /*
+         * @fn Print
+         * @brief prints vec
+        */
+        inline void Print()const  {
```
Comment (BIGbadEL) [inline src/math/vec2.h]: we should overload operator "<<" it would be more handy

File: src/math/vec2.h [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+            std::cout << "[" << m_x << ", " << m_y << "]" << std::endl;
+        }
+
+        /*
+         * @fn operator+
+         * @brief adds another vector to the current vector
+         * @param [const Vec2&] other vector
+        */
+        [[nodiscard]] constexpr Vec2& operator+(const Vec2& other)noexcept;
```
Comment (BIGbadEL) [inline src/math/vec2.h]: this has to be defined in this file

File: src/math/vec2.h [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+
+        /*
+         * @fn operator*
+         * @brief multiplies current vector by the passed multiplier
+         * @param [double] multiplier
+        */
+        [[nodiscard]] constexpr Vec2& operator*(double multiplier)noexcept;
```
Comment (BIGbadEL) [inline src/math/vec2.h]: this has to be defined in this file

File: src/math/vec2.h [hunk 18]
```diff
@@ -0,0 +1,95 @@
...
+         * @return [double] scalar product
+        */
+        double ScalarProduct(const Vec2& other)const;
+
+
+
+        // [[nodiscard]] constexpr Vec2 VectorProduct(const Vec2& other)noexcept;
+
+    private:
+        double m_x;
```
Comment (BIGbadEL) [inline src/math/vec2.h]: we should store floats

File: src/math/vec2.h [hunk 19]
```diff
@@ -0,0 +1,95 @@
...
+        double m_y;
```
Comment (BIGbadEL) [inline src/math/vec2.h]: we should store floats

**Review by BIGbadEL — CHANGES_REQUESTED** [review]

File: tests/test_vector2.cpp [hunk 20]
```diff
@@ -0,0 +1,28 @@
+//
+// Created by ajer on 16.12.18.
+//
+
+#include "pch.h"
+#include "gtest/gtest.h"
+#include "math/vec2.h"
+
+
```
Comment (BIGbadEL) [inline tests/test_vector2.cpp:9]: you should add more tests

File: src/math/vec2.h [hunk 21]
```diff
@@ -0,0 +1,95 @@
...
+         * @return [double] length
+        */
+        double Len()const;
+        /*
+         * @fn ScalarProduct
+         * @brief returns scalar product
+         * @param [const Vec2&] other
+         * @return [double] scalar product
+        */
+        double ScalarProduct(const Vec2& other)const;
```
Comment (Blinkuu) [inline src/math/vec2.h]: i think DotProduct() would be more suitable name

File: src/math/vec2.cpp (hunk 4 above)
Comment (4jeR) [reply src/math/vec2.cpp:28]: done

File: tests/test_vector2.cpp (hunk 20 above)
Comment (4jeR) [reply tests/test_vector2.cpp:9]: done

File: src/math/vec2.cpp [hunk 22]
```diff
@@ -0,0 +1,76 @@
...
+
+    Vec2 Vec2::operator+(const Vec2& other)const noexcept{
+        return Vec2(m_x + other.m_x, m_y + other.m_y);
+    }
+
+    Vec2 Vec2::operator*(float multiplier) const noexcept{
+        return Vec2(m_x * multiplier, m_y * multiplier);
+    }
+
+    Vec2& Vec2::operator=(const Vec2& other)noexcept{
```
Comment (BIGbadEL) [inline src/math/vec2.cpp]: it's unnecessary

File: src/math/vec2.cpp [hunk 23]
```diff
@@ -0,0 +1,76 @@
...
+        if(this != &other){
+            m_x = other.m_x;
+            m_y = other.m_y;
+        }
+        return *this;
+    }
+
+    bool Vec2::operator==(const Vec2& other) const noexcept{
```
Comment (BIGbadEL) [inline src/math/vec2.cpp]: it should be return only function

File: src/math/vec2.cpp [hunk 24]
```diff
@@ -0,0 +1,76 @@
...
+        if(m_x == other.m_x && m_y == other.m_y) return true;
+        return false;
+    }
+
+
+    Vec2 Vec2::CreateUnitX(){
```
Comment (BIGbadEL) [inline src/math/vec2.cpp]: there should be universal function that  normalize vector

File: src/math/vec2.cpp [hunk 25]
```diff
@@ -0,0 +1,76 @@
...
+        out << "[" << vec.m_x << ", " << vec.m_y << "]" << std::endl;
+        return out;
+    }
+
+    float Vec2::AngleBetween(const Vec2& other)const noexcept{
+
+        float dot = DotProduct(other);
+        float len = Len() * other.Len();
+        float radians = static_cast<float> (acos(dot/len));
+        return ( radians * 180.0f ) / static_cast<float>(M_PI);
```
Comment (BIGbadEL) [inline src/math/vec2.cpp]: it should return angle in radians

File: src/math/vec2.h (hunk 13 above)
Comment (BIGbadEL) [reply src/math/vec2.h]: ...

File: src/math/vec2.h [hunk 26]
```diff
@@ -0,0 +1,155 @@
...
+         * @brief returns m_y variable
+         * @return [float] m_y
+        */
+        [[nodiscard]] constexpr inline float GetY()const noexcept{return m_y;}
+
+        /*
+         * @fn Print
+         * @brief prints vec
+         */
+        inline void Print()const{std::cout << "[" << m_x << ", " << m_y << "]" << std::endl;}
```
Comment (BIGbadEL) [inline src/math/vec2.h]: we have two functions that do exacly the same thing

File: tests/test_vector2.cpp [hunk 27]
```diff
@@ -0,0 +1,119 @@
...
+    Bald::Math::Vec2 B = A.Reverse();
+
+
+    EXPECT_EQ(Bald::Math::Vec2(-3.5f, 2.0f) , B);
+}
+
+
+// TODO: Angle between vectors test
+
+//TEST(Angle, AngleBetweenTwoVectors) { //NOLINT
```
Comment (BIGbadEL) [inline tests/test_vector2.cpp]: xDDD

File: tests/test_vector2.cpp (hunk 20 above)
Comment (BIGbadEL) [reply tests/test_vector2.cpp:9]: what about +=, *=

**Review by BIGbadEL — CHANGES_REQUESTED** [review]

File: src/math/vec2.cpp (hunk 3 above)
Comment (4jeR) [reply src/math/vec2.cpp]: done

File: src/math/vec2.cpp (hunk 2 above)
Comment (4jeR) [reply src/math/vec2.cpp]: done

File: src/math/vec2.cpp (hunk 5 above)
Comment (4jeR) [reply src/math/vec2.cpp]: done

File: src/math/vec2.cpp (hunk 6 above)
Comment (4jeR) [reply src/math/vec2.cpp]: done

File: src/math/vec2.h (hunk 15 above)
Comment (4jeR) [reply src/math/vec2.h]: done

File: src/math/vec2.h (hunk 18 above)
Comment (4jeR) [reply src/math/vec2.h]: done

File: src/math/vec2.h (hunk 19 above)
Comment (4jeR) [reply src/math/vec2.h]: done

File: src/math/vec2.h (hunk 21 above)
Comment (4jeR) [reply src/math/vec2.h]: done

File: src/math/vec2.cpp (hunk 7 above)
Comment (4jeR) [reply src/math/vec2.cpp:68]: done

File: tests/test_vector2.cpp (hunk 20 above)
Comment (4jeR) [reply tests/test_vector2.cpp:9]: done

File: src/math/vec2.cpp (hunk 22 above)
Comment (4jeR) [reply src/math/vec2.cpp]: done

File: src/math/vec2.cpp (hunk 23 above)
Comment (4jeR) [reply src/math/vec2.cpp]: done

File: src/math/vec2.cpp (hunk 24 above)
Comment (4jeR) [reply src/math/vec2.cpp]: done

File: src/math/vec2.cpp (hunk 25 above)
Comment (4jeR) [reply src/math/vec2.cpp]: done

File: src/math/vec2.h (hunk 26 above)
Comment (4jeR) [reply src/math/vec2.h]: fixed

File: tests/test_vector2.cpp (hunk 27 above)
Comment (4jeR) [reply tests/test_vector2.cpp]: XD

Comment (4jeR) [reply tests/test_vector2.cpp]: DONE

File: src/math/vec2.h [hunk 28]
```diff
@@ -0,0 +1,181 @@
...
+        Vec2(float x = 0.0f, float y = 0.0f)
+                :m_x(x), m_y(y){}
+
+
+        /*
+         * @fn MakeUnitVec
+         * @brief creates unit vector with the length of 1.0f
+         * @param [const Vec2&] vec -> current vector
+         * @return [Vec2] unit long vector
+         * */
```
Comment (piotrmoszkowicz) [inline src/math/vec2.h]: Fix formatting

File: src/math/vec2.h [hunk 29]
```diff
@@ -0,0 +1,181 @@
...
+        [[nodiscard]] static Vec2 MakeUnitVec(const Vec2& vec)noexcept;
+
+
+        /*
+         * @fn Normalize
+         * @brief normalizes the vector to the length of 1.0f
+         * @return [Vec2&] unit long vector
+         * */
```
Comment (piotrmoszkowicz) [inline src/math/vec2.h]: Fix formatting

File: src/math/vec2.h [hunk 30]
```diff
@@ -0,0 +1,181 @@
...
+         * @fn operator+
+         * @brief adds another vector to the current vector
+         * @param [const Vec2&] other vector
+        */
+        [[nodiscard]] Vec2 operator+(const Vec2& other)const noexcept;
+
+
+        /*
+         * @fn operator-
+         * @brief adds another vector to the current vector
```
Comment (piotrmoszkowicz) [inline src/math/vec2.h]: As far as I know it subtracts. I know, we can argue that subtraction is kind of addition, but let's change it :)

File: src/math/vec2.h [hunk 31]
```diff
@@ -0,0 +1,181 @@
...
+         * @fn operator==
+         * @brief checks if the two vectors are the same
+         * @param [const Vec&] other Vec2
+        */
+        [[nodiscard]] bool operator==(const Vec2& other) const noexcept;
+
+
+        /*
+         * @fn operator!=
+         * @brief checks if the two vectors are the same
```
Comment (piotrmoszkowicz) [inline src/math/vec2.h]: Aren't*

File: src/math/vec2.h [hunk 32]
```diff
@@ -0,0 +1,181 @@
...
+         * @brief returns length of the vector
+         * @return [float] length
+        */
+        [[nodiscard]] float Len()const noexcept;
+
+
+        /*
+         * @fn DotProduct
+         * @brief returns scalar product
+         * @param [const Vec2&] other
```
Comment (piotrmoszkowicz) [inline src/math/vec2.h]: A bit better explanation of param - let's say other vector at least

File: src/math/vec2.cpp [hunk 33]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+
+    float Vec2::AngleBetween(const Vec2& other)const noexcept{
+
+        float dot = DotProduct(other);
+        float len = Len() * other.Len();
+        float radians = static_cast<float> (acos(dot/len));
+
+        // degrees -> (radians * 180.0f ) / static_cast<float>(M_PI);
```
Comment (piotrmoszkowicz) [inline src/math/vec2.cpp]: If it's not needed remove it before merge.

File: tests/test_vector2.cpp [hunk 34]
```diff
@@ -0,0 +1,165 @@
...
+
+    EXPECT_EQ(-23.33f, A.GetY());
+}
+
+
+
+TEST(MakeUnitVec, StaticUnitVec) { //NOLINT
+    Bald::Math::Vec2 A = Bald::Math::Vec2::MakeUnitVec(Bald::Math::Vec2(3.0f, 4.0f));
+
+    EXPECT_EQ(1.0f, A.Len());
```
Comment (piotrmoszkowicz) [inline tests/test_vector2.cpp]: It should test that vector has correct X and Y, not it's length.

File: tests/test_vector2.cpp [hunk 35]
```diff
@@ -0,0 +1,165 @@
...
+}
+
+
+
+TEST(NormalizeVector, UnitVecMethod) { //NOLINT
+    Bald::Math::Vec2 A(5.0f,12.0f);
+
+    A.Normalize();
+
+    EXPECT_EQ(1.0f, A.Len());
```
Comment (piotrmoszkowicz) [inline tests/test_vector2.cpp]: It should test that vector has correct X and Y, not it's length.

File: tests/test_vector2.cpp [hunk 36]
```diff
@@ -0,0 +1,165 @@
...
+TEST(EqualsEqualsOperator, ComparingOperator) { //NOLINT
+    Bald::Math::Vec2 A(2.0f,-3.0f);
+    Bald::Math::Vec2 B(2.0f,-3.0f);
+
+    EXPECT_EQ(true, A == B);
+}
+
+
+
+TEST(NotEqualsOperator2, ComparingOperator2) { //NOLINT
```
Comment (piotrmoszkowicz) [inline tests/test_vector2.cpp]: No need to add 2 after ComparingOperator. 
These names work like this:
TEST(NameOfAllOfTestScenarios, SingleScenarioName)

For example here it simply should be TEST(NotEqualsOperator, DifferentClassInstances)

I would personally also add tests, which expect to get false.

File: tests/test_vector2.cpp [hunk 37]
```diff
@@ -0,0 +1,165 @@
...
+}
+
+
+
+TEST(Reverse, ReversingTheVec) { //NOLINT
+    Bald::Math::Vec2 A(3.5f, -2.0f);
+
+    Bald::Math::Vec2 B = A.Reverse();
+
+    EXPECT_EQ(Bald::Math::Vec2(-3.5f, 2.0f) , B);
```
Comment (piotrmoszkowicz) [inline tests/test_vector2.cpp]: Afaik, you can move A.Reverse() into EXPECT_EQ instead of using one more variable.

File: tests/test_vector2.cpp [hunk 38]
```diff
@@ -0,0 +1,165 @@
...
+
+    EXPECT_EQ(true, A != B);
+}
+
+
+
+TEST(Length, CalculateLength) { //NOLINT
+    Bald::Math::Vec2 A(3.0f, 4.0f);
+
+    EXPECT_EQ(5.0f, A.Len());
```
Comment (piotrmoszkowicz) [inline tests/test_vector2.cpp]: ASSERT_FLOAT_EQ

File: tests/test_vector2.cpp [hunk 39]
```diff
@@ -0,0 +1,165 @@
...
+}
+
+
+
+TEST(DotProduct, CalculateScalarProduct) { //NOLINT
+    Bald::Math::Vec2 A(1.0f, 1.0f);
+    Bald::Math::Vec2 B = Bald::Math::Vec2(2.0f, 2.0f);
+
+    EXPECT_EQ(4.0f, A.DotProduct(B));
```
Comment (piotrmoszkowicz) [inline tests/test_vector2.cpp]: ASSERT_FLOAT_EQ

File: tests/test_vector2.cpp [hunk 40]
```diff
@@ -0,0 +1,165 @@
...
+    EXPECT_EQ(Bald::Math::Vec2(8.0f,-12.0f), A);
+}
+
+
+
+TEST(EqualsEqualsOperator, ComparingOperator) { //NOLINT
+    Bald::Math::Vec2 A(2.0f,-3.0f);
+    Bald::Math::Vec2 B(2.0f,-3.0f);
+
+    EXPECT_EQ(true, A == B);
```
Comment (piotrmoszkowicz) [inline tests/test_vector2.cpp]: Change to ASSERT_TRUE(A == B)

File: tests/test_vector2.cpp [hunk 41]
```diff
@@ -0,0 +1,165 @@
...
+}
+
+
+
+TEST(NotEqualsOperator2, ComparingOperator2) { //NOLINT
+    Bald::Math::Vec2 A(2.0f,-3.0f);
+    Bald::Math::Vec2 B(4.0f,-11.2f);
+
+    EXPECT_EQ(true, A != B);
```
Comment (piotrmoszkowicz) [inline tests/test_vector2.cpp]: Change to ASSERT_TRUE

File: tests/test_vector2.cpp [hunk 42]
```diff
@@ -0,0 +1,165 @@
...
+
+
+
+TEST(MinusEqualsOperator, SubtractingOperator) { //NOLINT
+    Bald::Math::Vec2 A(2.0f, 4.0f);
+    Bald::Math::Vec2 B(1.0f, 1.0f);
+
+    A -= B;
+
+    EXPECT_EQ(Bald::Math::Vec2(1.0f,3.0f), A);
```
Comment (piotrmoszkowicz) [inline tests/test_vector2.cpp]: Why not move it into the EXPECT_EQ -> EXPECT_EQ(..., A-=B)

File: tests/test_vector2.cpp [hunk 43]
```diff
@@ -0,0 +1,165 @@
...
+}
+
+
+
+TEST(MultiEqualsOperator, MultiplyingOperator) { //NOLINT
+    Bald::Math::Vec2 A(2.0f,-3.0f);
+
+    A *= 4.0f;
+
+    EXPECT_EQ(Bald::Math::Vec2(8.0f,-12.0f), A);
```
Comment (piotrmoszkowicz) [inline tests/test_vector2.cpp]: Move *=

File: tests/test_vector2.cpp [hunk 44]
```diff
@@ -0,0 +1,165 @@
...
+
+
+
+TEST(PlusEqualsOperator, AddingOperator) { //NOLINT
+    Bald::Math::Vec2 A(2.0f, 4.0f);
+    Bald::Math::Vec2 B(1.0f, 1.0f);
+
+    A += B;
+
+    EXPECT_EQ(Bald::Math::Vec2(3.0f,5.0f), A);
```
Comment (piotrmoszkowicz) [inline tests/test_vector2.cpp]: Move +=

File: tests/test_vector2.cpp [hunk 45]
```diff
@@ -0,0 +1,165 @@
...
+}
+
+
+
+TEST(MultiplyOperator, MultiplyingOperator) { //NOLINT
+    Bald::Math::Vec2 A(2.0f,-4.0f);
+
+    Bald::Math::Vec2 C(A*3.0f);
+
+    EXPECT_EQ(Bald::Math::Vec2(6.0f,-12.0f), C);
```
Comment (piotrmoszkowicz) [inline tests/test_vector2.cpp]: Move A*3.0f

File: tests/test_vector2.cpp [hunk 46]
```diff
@@ -0,0 +1,165 @@
...
+
+
+
+TEST(MinusOperator, SubtractingOperator) { //NOLINT
+    Bald::Math::Vec2 A(2.0f,-4.0f);
+    Bald::Math::Vec2 B(1.0f, 3.0f);
+
+    Bald::Math::Vec2 C(A-B);
+
+    EXPECT_EQ(Bald::Math::Vec2(1.0f,-7.0f), C);
```
Comment (piotrmoszkowicz) [inline tests/test_vector2.cpp]: Move A-B

File: tests/test_vector2.cpp [hunk 47]
```diff
@@ -0,0 +1,165 @@
...
+
+
+
+TEST(PlusOperator, AddingOperator) { //NOLINT
+    Bald::Math::Vec2 A(2.0f,-4.0f);
+    Bald::Math::Vec2 B(1.0f, 3.0f);
+
+    Bald::Math::Vec2 C(A+B);
+
+    EXPECT_EQ(Bald::Math::Vec2(3.0f,-1.0f), C);
```
Comment (piotrmoszkowicz) [inline tests/test_vector2.cpp]: Move A+B

File: tests/test_vector2.cpp [hunk 48]
```diff
@@ -0,0 +1,165 @@
...
+
+    EXPECT_EQ(66.0f, A.GetX());
+}
+
+
+
+TEST(GetterY, GetY) { //NOLINT
+    Bald::Math::Vec2 A(2.0f, -23.33f);
+
+    EXPECT_EQ(-23.33f, A.GetY());
```
Comment (piotrmoszkowicz) [inline tests/test_vector2.cpp]: ASSERT_FLOAT_EQ

File: tests/test_vector2.cpp [hunk 49]
```diff
@@ -0,0 +1,165 @@
...
+#include "pch.h"
+#include "gtest/gtest.h"
+#include "math/vec2.h"
+
+
+
+TEST(GetterX, GetX) { //NOLINT
+    Bald::Math::Vec2 A(66.0f,3.4f);
+
+    EXPECT_EQ(66.0f, A.GetX());
```
Comment (piotrmoszkowicz) [inline tests/test_vector2.cpp]: ASSERT_FLOAT_EQ

**Review by piotrmoszkowicz — CHANGES_REQUESTED** [review]
Before all change make reformat code on all new files (CTRL + ALT + L when file is opened). It should add all the spaces and fix some formatting bugs.

File: src/math/vec2.h (hunk 28 above)
Comment (4jeR) [reply src/math/vec2.h]: done

File: src/math/vec2.h (hunk 29 above)
Comment (4jeR) [reply src/math/vec2.h]: done

File: src/math/vec2.h (hunk 30 above)
Comment (4jeR) [reply src/math/vec2.h]: done

File: src/math/vec2.h (hunk 31 above)
Comment (4jeR) [reply src/math/vec2.h]: done

File: src/math/vec2.h (hunk 32 above)
Comment (4jeR) [reply src/math/vec2.h]: done

File: tests/test_vector2.cpp (hunk 45 above)
Comment (4jeR) [reply tests/test_vector2.cpp]: done

File: tests/test_vector2.cpp (hunk 46 above)
Comment (4jeR) [reply tests/test_vector2.cpp]: done

File: tests/test_vector2.cpp (hunk 47 above)
Comment (4jeR) [reply tests/test_vector2.cpp]: done

File: tests/test_vector2.cpp (hunk 48 above)
Comment (4jeR) [reply tests/test_vector2.cpp]: done

File: tests/test_vector2.cpp (hunk 49 above)
Comment (4jeR) [reply tests/test_vector2.cpp]: done

File: src/math/vec2.h (hunk 28 above)
Comment (piotrmoszkowicz) [reply src/math/vec2.h]: Very last thing here - last line doesn't need two * and should have a bit different indent

File: src/math/vec2.h (hunk 29 above)
Comment (piotrmoszkowicz) [reply src/math/vec2.h]: Same as above

File: src/math/vec2.h (hunk 28 above)
Comment (4jeR) [reply src/math/vec2.h]: done

**Review by piotrmoszkowicz — APPROVED** [review]
Very good job! Let's wait for @BIGbadEL decision and we can merge it :)

File: src/math/vec2.cpp [hunk 50]
```diff
@@ -0,0 +1,102 @@
...
+        if (len != 0.0f) {
+            m_x /= len;
+            m_y /= len;
+        }
+        return *this;
+    }
+
+
+    float Vec2::Len() const noexcept {
+        return static_cast<float>(sqrt(pow(m_x, 2.0f) + pow(m_y, 2.0f)));
```
Comment (BIGbadEL) [inline src/math/vec2.cpp]: function pow is very slow it this case simple multiplication will be better

**Review by BIGbadEL — COMMENTED** [review]
it's good, now what left is optimization. Try to make everything that's possible constexpr

File: src/math/vec2.cpp (hunk 50 above)
Comment (4jeR) [reply src/math/vec2.cpp]: done

**Review by BIGbadEL — APPROVED** [review]

**Pull request merged**